### DISCUSSION TITLE
How to disable telemetry in config file?

### DISCUSSION THREAD

**Discussion by deadnews** [grafana-cold-storage/agent#2539] [Q&A]
https://github.com/grafana/agent/discussions/1777

> This is release `v0.25.0` of the Grafana Agent.
>     * As of this release, the Grafana Agent will send anonymous, but uniquely-identifiable information around enabled feature flags to `stats.grafana.org`.  Statistics help us better understand how Grafana Agent is used. This helps us prioritize features and documentation. If you would like to disable the reporting, Grafana Agent provides the flag `-disable-reporting`
>       to stop the reporting.

How to disable telemetry via `--config.file`, in `/etc/grafana-agent/grafana-agent.yml`?

**Comment by rfratto** [comment]
Hi, there is currently no way to disable reporting in the config file and it must be done via the `-disable-reporting` command line flag mentioned above.

**Reply by Sommerwiesel** [reply]
Just found this - sorry for thread necro but this is a really bad practise. Any kind of analytics should be, inaccordance with gdpr, OPT-IN and not enabled by default, especially if disabling it is impossible via config file and requires overwriting the systemd unit. I just found out that my server has been sending this kind of analytics without my knowledge and this is in my opinion a huge screw up on your part.
Please comply with gdpr as well as some common human sense and change this in a future update of the grafana agent so that its disabled by default and opt-in only.

**Reply by vnugent** [reply]
@rfratto Any updates on this?

**Reply by johnaAr555** [reply]
**_bye_**

**Reply by ocontant** [reply]
#### This response should have been under the Tempo repository, during my search I crossed over to this thread and assumed it was under Tempo Discussion. I plan to move it under the tempo repo in the next couple hours. 

Please provide your feedback if it should remain here, perhaps through association or other context where my feedback is pertinent to the grafana agent as well.


### @grafana-dylan @grafana-dee @grafana-eddie — please escalate this to your legal and compliance leadership immediately.

> Grafana has been approved for use in sensitive infrastructure, including within the U.S. Department of Defense (per your own [public post](https://grafana.com/blog/2022/02/04/a-look-at-how-the-u.s.-department-of-defense-deploys-the-grafana-stack/)). 

**With that level of trust, your responsibility for data handling and privacy is not just technical—it's legal, ethical, and geopolitical.**

-----------------------

### Enabling telemetry by default, without a transparent, clearly documented, and easily configurable way to disable it (from both configuration files and command-line options), is deeply problematic.

This practice is:
    - Unethical: It violates the principle of informed consent, especially when telemetry is bundled into software designed to run in secure and regulated environments.**

    - Legally questionable: In many jurisdictions (e.g., GDPR, CCPA), collecting data without explicit user consent can constitute a violation—even if that data is anonymized. The notion that "anonymized telemetry" is risk-free is outdated and disproven. Aggregated or correlated data can be deanonymized using modern techniques, especially when network or behavioral patterns are involved.

   - Professionally irresponsible: It erodes trust in open source observability tools, particularly when users discover this behaviour only after extensive investigation or monitoring.


> To put this in perspective: If a Chinese or Russian company embedded default telemetry in software adopted by foreign critical infrastructure, it would be perceived as a national security threat. The same scrutiny must apply here.


--------------------------------

This issue has been raised repeatedly since at least 2023, and the lack of resolution reflects poorly on Grafana's commitment to user trust and transparency.

**Immediate steps I urge you to take:**

    - Disable telemetry by default in your next release.
    - Publish clear, accessible documentation on how to fully disable telemetry via config and CLI flags.
    - Provide an explicit changelog note acknowledging this change, with a rationale.

-----------------------------------

**For context**: my own infrastructure blocks all outbound traffic by default, which prevented any data leakage. Had this not been the case, this issue would already be in the hands of our legal team.

**Please understand:** no data is truly anonymous when aggregated. If you operate in regulated, government, or security-sensitive environments, **you already know this. Let’s not pretend otherwise.**


**This is your opportunity to do the right thing—transparently and decisively.**

**Reply by mvtab** [reply]
I once did a pull request at cilium because of this. In the end it's pretty easy, if by telemetry you mean analytics: https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/ scroll down (or search for) to `[analytics]`. I had done some Kubernetes tests with Cilium a year ago or so and disabling the analytics made Grafana stop sending requests to the outside world.

Might have changed in the meantime, since the whole Grafana Landscape is on parkour-mode for 1-2 years.

Update:
I completely agree with @ocontant, legally this default needs to be `false`.

**Reply by ocontant** [reply]
Hi mvtab, I will tag you when I move the thread under the Tempo repository discussion. I would welcome your feedback under that thread, as it will be where traction should be targeted.

**Reply by mvtab** [reply]
I'll just drop this here for now

https://github.com/cilium/cilium/security/advisories/GHSA-mxv3-27mh-wcfj
https://github.com/cilium/cilium/pull/31989/commits/ceed3923d1c69a61182ee8f0cfbd432d9ef976a6

**Reply by ocontant** [reply]
Your link return a 404, maybe the url is incomplete?

New discussion started in Tempo repo: https://github.com/grafana/tempo/discussions/5001

**Reply by Sommerwiesel** [reply]
@ocontant Good luck with that (and I mean that). I already tried to convince them to change this practise, here: https://github.com/grafana/alloy/issues/317

**Reply by ocontant** [reply]
Thanks for the reference https://github.com/Sommerwiesel

I read the topic and Grafana is wrong all the way. Anonymized data is never anonymous when aggregated, and this narrative is just the same argument used by the Advertiser to defend an unethical and immoral position. 

From the moment it is enabled by default this is straight malicious and goes against Nettiquette that existed before mass-surveillance. 

There is just no justification whatsoever to collect data without explicit authorization of users, and any attempt to hide it or obfuscate it, and it is automatic obfuscation when you don't provide a splash screen to warn the user that their data is being sent out. 

For me, this is just a reason to fork Grafana and respin it under another name. Sad, but if founder and developers are acting malevolently, this is what will happen in the end. And the only reason, this hasn't happened yet, it's because barely anyone knows about it, as it is obfuscated well enough.

If the only reference is the last part of a documentation page, you can be sure no one ever read it. No one read documentation end to end in details. We scan and surf for what we are looking for.

There is no reference in the yaml config file, which would be the ONLY place where anyone deploying Grafana is very likely to check and see.  SystemD config file, no lambda user ever check there. If you are not a savvy Linux administrator, you will not even know it exist.

In my honest opinion, the whole argument is bullshit and dishonest.

**Reply by ocontant** [reply]
I'm opening an issue on the grafana/grafana repo, seems to me it seems the right place to raise this properly, as grafana/grafana is logically the central focal point.

**Comment by ocontant** [comment]
I opened an issue on grafana/grafana : https://github.com/grafana/grafana/issues/104077

Feel free to bring your support and feedback!
